### DISCUSSION TITLE
Raw查询如何使用in条件？

### DISCUSSION THREAD

**Discussion by yc-go** [rbatis/rbatis#360] [General]
使用rb.fetch()方法时，如何设置in对应的参数？

**Comment by zhuxiujia** [comment]
> 使用rb.fetch()方法时，如何设置in对应的参数？

rb.fetch() 是未经过处理的sql直接执行，所以你需要处理好in再传入。
或者，考虑使用htmlsql/pysql/crud!   可以使用 编译形式的处理
```
use rbatis::sql::IntoSql;
use rbs::value::map::ValueMap;
impl_select!(BizActivity{select_by_method(ids:&[&str],logic:ValueMap) -> Option => "`where ${logic.sql()} and id in ${ids.sql()}   limit 1`"});
```

**Comment by yc-go** [comment]
意思是我不能通过 in (?) 再匹配 Value这样的方式来使用吗？只能自己把 sql 处理成直接的如 in ('foo','bar')  这样的形式？

**Comment by zhuxiujia** [comment]
> 意思是我不能通过 in (?) 再匹配 Value这样的方式来使用吗？只能自己把 sql 处理成直接的如 in ('foo','bar') 这样的形式？

可以使用例如
```
rb.fetch("select table where id in (?,?,?)",vec![1.into(),2.into(),3.into()]).await;
```

**Comment by yc-go** [comment]
> > 意思是我不能通过 in (?) 再匹配 Value这样的方式来使用吗？只能自己把 sql 处理成直接的如 in ('foo','bar') 这样的形式？
> 
> 可以使用例如
> 
> ```
> rb.fetch("select table where id in (?,?,?)",vec![1.into(),2.into(),3.into()]).await;
> ```

好的，谢谢您，这样子应该能解决我的问题了
